src/result_collection.jl

```diff
@@ -75,15 +77,18 @@ function collect_results!(filename, folder;
     subfolders = false,
     rpath = nothing,
     verbose = true,
+    update = false,
     newfile = false, # keyword only for defining collect_results without !
     kwargs...)
 
     if newfile || !isfile(filename)
         !newfile && verbose && @info "Starting a new result collection..."
         df = DataFrames.DataFrame()
+        mtime_df = 0.0
     else
         verbose && @info "Loading existing result collection..."
         df = wload(filename)["df"]
+        mtime_df = mtime(filename)
```

Comment:
Hm, this appears to be rather brittle to me.

Suppose you 
- add and modify files
- then call `collect_results` without the `update` kw
- then remember and call it again with `update`

In this scenario, your updated files never make it into the `DataFrame`.
It may be more robust to attach `mtime` attributes for all data files individually.

Reply:
This is a good point. I considered adding an `mtime` field (or even better a hash of the file, that'd be even more reliable). But this would not work for files created by `collect_results!` before this was implemented. And I wanted the current behaviour to be preserved for `update=false`.

I guess we could check against the database's `mtime` if no `mtime` or `hash` field is there, issuing a warning and then creating the field?

Reply:
I'm wondering, whether it makes sense to include the `mtime` or `hash` or whatever in the *user-facing* `DataFrame`.
One could alternatively carry it around separately. That would keep the field more of an implementation detail .I'm not sure that's useful though.

Reply:
I've now put the mtime it in the user-facing `DataFrame`. But we could put it as a `Dict` with `path` => `mtime` under an additional `mtime` key in the saved "results_*.jld2" file instead of the `DataFrame`. That might even simplify the logic.

Reply:
I think I'd prefer that